driver/pool_driver.hpp

```diff
@@ -137,29 +136,43 @@ int PoolDriver_impl<Tgpu, Tref, Index>::ParseCmdLineArgs(int argc, char* argv[])
 template <typename Tgpu, typename Tref, typename Index>
 int PoolDriver_impl<Tgpu, Tref, Index>::GetandSetData()
 {
-    spatial_dim             = inflags.GetValueInt("spatial_dim");
-    std::vector<int> in_len = GetInputTensorLengthsFromCmdLine();
+    auto input  = inflags.GetValueTensor("input");
+    spatial_dim = input.lengths.size() - 2;
+
+    if(input.SetTensordDescriptor(inputTensor, data_type) != miopenStatusSuccess)
+        MIOPEN_THROW("Error parsing input tensor: " + inflags.GetValueStr("input") + ".");
+
+    auto dinput = inflags.GetValueTensor("dinput").FillMissing(input);
```

Comment:
Why do we need a special `dinput` flag? Cant we reuse the `input` flag for `dinput` etc ?

Reply:
To allow passing any values to the API via the driver. It is not needed for valid configurations but allows bizarre case testing. More flexibility of the driver allows more experimentation when reproducing issues. I can remove this, but it allows easy access to providing incorrect data. Also it is initialized from `input` if no value provided, so it may be ignored when not neeeded.

Reply:
Strictly speaking, we don't need it right now, but it may save some time later, one never knows.